src/App.js

```diff
@@ -12,134 +12,135 @@ import imgLoading from "./images/isLoading.gif";
-    const willSetToTrue = completedTodoNumber !== todos.length;
+    const countDone = Object.keys(todos)
+      .map((key) => todos[key])
+      .filter((todo) => todo.done).length;
 
-    const todoTobeUpdated = todos.filter(
-      (todo) => todo.done === !willSetToTrue
-    );
+    const doneValue = Object.keys(todos).length !== countDone;
 
-    for (let index = 0; index < todoTobeUpdated.length; index++) {
-      const todo = todoTobeUpdated[index];
-      axios.put(`${URL.TODOS}/${todo.id}`, {
-        content: todo.content,
-        done: !todo.done,
+    const newTodos = Object.keys(todos).reduce((acc, key) => {
+      return Object.assign({}, acc, {
+        [key]: { ...todos[key], done: doneValue },
       });
-    }
+    }, {});
 
     this.setState({
-      todos: todos.map((todo) => ({
-        ...todo,
-        done: willSetToTrue,
-      })),
+      todos: newTodos,
     });
   };
 
-  toggleTodo = (todo, id) => {
-    axios
-      .put(`${URL.TODOS}/${id}`, {
-        content: todo.content,
-        done: !todo.done,
-      })
-      .then(() => {
-        this.setState({
-          todos: this.state.todos.map((todo) => ({
-            ...todo,
-            done: todo.id === id ? !todo.done : todo.done,
-          })),
-        });
-      });
+  toggleTodo = (id, content, done) => {
+    const todoToggle = {
+      id,
+      content,
+      done: !done,
+    };
+
+    this.setState({
+      todos: Object.assign({}, this.state.todos, { [id]: todoToggle }),
+    });
   };
 
-  editTodo = (id, todo, content) => {
-    axios
-      .put(`${URL.TODOS}/${id}`, {
-        content: content,
-        done: todo.done,
-      })
-      .then(() => {
-        this.setState({
-          todos: this.state.todos.map((todo) => ({
-            ...todo,
-
-            content: todo.id === id ? content : todo.content,
-          })),
-        });
-      });
+  editTodo = ({ id, content, done }) => {
+    const todoEdit = {
+      id,
+      content,
+      done,
+    };
+
+    this.setState({
+      todos: Object.assign({}, this.state.todos, { [id]: todoEdit }),
+    });
   };
 
   deleteTodo = (id) => {
-    axios.delete(`${URL.TODOS}/${id}`).then(() => {
-      this.setState({
-        todos: this.state.todos.filter((todo) => todo.id !== id),
-      });
+    //dung reduce
+    const { todos } = this.state;
+    const willTodo = Object.keys(todos).reduce(() => {
+      return Object.assign(
+        {},
+        Object.keys(todos)
+          .map((key) => todos[key])
+          .filter((todo) => todo.id !== id)
+      );
+    }, {});
+
+    //dungf filrer
+    // const { todos } = this.state;
+    // const newTodo = Object.keys(todos)
+    //   .map((key) => todos[key])
+    //   .filter((todo) => todo.id !== id);
+    this.setState({
+      todos: Object.assign({}, willTodo),
     });
   };
 
-  updateStatus = (status) => {
+  clearCompleted = () => {
+    const { todos } = this.state;
+    const newTodo = Object.keys(todos).reduce(() => {
```

Comment:
does this function work?
check this again, I don't think it work

Reply:
it works